test/integration/tlspolicy_controller_test.go

```diff
@@ -322,11 +359,11 @@ var _ = Describe("TLSPolicy", Ordered, func() {
 					certList := &certmanv1.CertificateList{}
 					err := k8sClient.List(ctx, certList, &client.ListOptions{Namespace: testNamespace})
 					Expect(err).ToNot(HaveOccurred())
-					if len(certList.Items) != 2 {
-						return fmt.Errorf("expected CertificateList to be 2")
+					if len(certList.Items) != 3 {
+						return fmt.Errorf("expected CertificateList to be 3")
 					}
```

Comment:
This was originally intended to test multiple listeners, but intentionally had 2 with the same name to check it only created 2 even though there was 3 listeners. 

We probably in reality need more test cases that describe each case better and actually check the created certificates have the correct DNSNames for the expected listeners.

Reply:
ok, I'll put this one back how it was - I assumed it was a typo - and make a new context for my tests.